data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/configuration/KeyConfig.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.configuration;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.NotNull;
+
+import java.util.List;
+
+public class KeyConfig {
+
+    @JsonProperty("source")
+    @NotNull
+    String source;
+
+    @JsonProperty("target")
+    String target;
+
+    @JsonProperty("attributes")
```

Comment:
Should `attributes` be defined per key? This seems to negate a couple benefits from using the same processor:

* It requires additional configurations that could be shared.
* Will this prevent using caching in the same processor?